lib/srv/desktop/windows_server.go

```diff
@@ -333,6 +340,7 @@ func NewWindowsService(cfg WindowsServiceConfig) (*WindowsService, error) {
 		Clock: s.cfg.Clock,
 	})
 	if err != nil {
+		s.Close()
```

Comment:
This might be a good place to apply that "!ok/close" idiom:

```go
// Declare closeable resources
ok := false
defer func() {
  if !ok {
    // Close resources
  }
}()

// Method continues ...

ok = true
return
```

Reply:
I like it!